packages/webpack/src/runtime/runtime-namespace.ts

```diff
@@ -7,6 +7,7 @@ type DebugTools = {
 
 type RequiredProperty<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>
 export interface RuntimeNamespace {
+  scuttling: {scuttle: (globalThis: any, scuttleGlobalThis: LavaMoatPluginOptions["scuttleGlobalThis"]) => {}}
```

Comment:
you said it's supposed to return nothing?

```suggestion
  scuttling: {scuttle: (globalThis: any, scuttleGlobalThis: LavaMoatPluginOptions["scuttleGlobalThis"]) => void}
```

I don't like `any` here -- it should be `Record<string, unknown>` if it doesn't cause type errors elsewhere; otherwise `Record<string, any>` is more appropriate. (`string` because IIRC we don't care about `number | symbol` which are the _other_ types of properties that could live in any given object)

I'd _strongly suggest_ to do the following, so we aren't tempted to tightly couple the type of the `scuttleGlobalThis` arg to the `LavaMoatPluginOptions` type:

1. Rename `ScuttlerConfig` to `ScuttlerObjectConfig` or `ScuttlerConfigObject` or the like
2. Create new `type ScuttlerConfig = ScuttlerObjectConfig | boolean`
3. Instead of `LavaMoatPluginOptions["scuttleGlobalThis"]` above, use the new `ScuttlerConfig`

Reply:
like this? 5c7226de5

Reply:
like this? a40a28044

Reply:
wait, but isn't that `globalThis` THE `globalThis`? In that case, it's far from being a record and we're about to mess it up so bad no typescript server is gonna figure it out.

Reply:
what is it then? sorry for not fully getting ts yet

Reply:
if this is the atual reference to the realm global aka globalThis aka window, it's not a string-keyed record and trying to type it properly is a futile endeavor.